pkg/tbtc/chain.go

```diff
@@ -198,6 +215,56 @@ type HeartbeatRequestedEvent struct {
 	BlockNumber     uint64
 }
 
+// DepositRevealedEvent represents a deposit reveal event.
+type DepositRevealedEvent struct {
+	FundingTxHash       bitcoin.Hash
+	FundingOutputIndex  uint32
+	Depositor           chain.Address
+	Amount              uint64
+	BlindingFactor      [8]byte
+	WalletPublicKeyHash [20]byte
+	RefundPublicKeyHash [20]byte
+	RefundLocktime      [4]byte
+	Vault               *chain.Address
+	BlockNumber         uint64
+}
+
+func (dre *DepositRevealedEvent) unpack() *Deposit {
+	return &Deposit{
+		Utxo: &bitcoin.UnspentTransactionOutput{
+			Outpoint: &bitcoin.TransactionOutpoint{
+				TransactionHash: dre.FundingTxHash,
+				OutputIndex:     dre.FundingOutputIndex,
+			},
+			Value: int64(dre.Amount),
+		},
+		Depositor:           dre.Depositor,
+		BlindingFactor:      dre.BlindingFactor,
+		WalletPublicKeyHash: dre.WalletPublicKeyHash,
+		RefundPublicKeyHash: dre.RefundPublicKeyHash,
+		RefundLocktime:      dre.RefundLocktime,
+		Vault:               dre.Vault,
+	}
+}
+
+// DepositRevealedEventFilter is a component allowing to filter DepositRevealedEvent.
+type DepositRevealedEventFilter struct {
+	StartBlock          uint64
+	EndBlock            *uint64
+	Depositor           []chain.Address
+	WalletPublicKeyHash [][20]byte
+}
+
+// DepositChainRequest represents a deposit request stored on-chain.
+type DepositChainRequest struct {
```

Comment:
What do you think about renaming it to `RevealedDeposit`?

I know it is `Deposit.DepositRequest` in Solidity but from the time perspective, I am not sure why we named it that way.

Reply:
I wanted to strongly emphasize that this is a chain-related struct and that it represents deposit data recorded on-chain. We already have a `tbtc.Deposit` struct which represents a deposit in this package. Adding `tbtc.RevealedDeposit` will cause confusion. 

I think the reasoning behind this naming was:
- A "deposit" is something actually made on the BTC chain. On the client side, it is represented by `tbtc.Deposit` and it contains fields required to reconstruct the script hash and data about the funding tx.
- a "depositRequest" is kind of "hey Bridge, I made this deposit on BTC chain, please handle it on your side".  On the client side it is represented by `tbtc.DepositChainRequest` and contains bridge-specific info like: when it was revealed, when it was swept, etc.

Reply:
That makes sense. Can we just extend the documentation of `DepositChainRequest` a bit:

```
// DepositChainRequest represents a deposit request stored on-chain.
// This is a deposit revealed to the Bridge and recorded on-chain. There is no
// guarantee this deposit actually happened on the Bitcoin side.
type DepositChainRequest struct {
```

?

Reply:
See 80b6a3e9